src/Discord/Parts/Channel/Message.php

```diff
@@ -79,6 +80,125 @@ public function reply($text)
     {
         return $this->channel->sendMessage("{$this->author}, {$text}");
     }
+    
+    /**
+     * React to a message.
+     *
+     * @param Emoji $emoticon  The emoticon to react with. (example: '👎', custom: ':michael:251127796439449631')
+     *
+     * @return \React\Promise\Promise
+     */
+    public function react($emoticon)
+    {
+        $deferred = new Deferred();
+
+        $this->http->put(
+            "channels/{$this->channel->id}/messages/{$this->id}/reactions/{$emoticon}/@me"
+        )->then(
+            function ($response) use ($deferred) {
+
+                $deferred->resolve($this);
+            },
+            \React\Partial\bind_right($this->reject, $deferred)
+        );
+
+        return $deferred->promise();
+    }
+    
+    /**
+     * Delete a reaction.
+     *
+     * @param array $settings  (id and emoticon)
+     *
+     * @return \React\Promise\Promise
+     */
+    public function react_delete($type, $emoticon = null, $id = null)
```

Comment:
I will change this to `deleteReaction` to match the function name style of the rest of the library. Also changing the `types` to constants.